ReloadingDatabasesWhilePerformingGeoLookupsIT#test failed

**Build scan**: https://gradle-enterprise.elastic.co/s/4kvggmablz2ea/console-log?task=:modules:ingest-geoip:internalClusterTest

**Repro line**: `REPRODUCE WITH: ./gradlew ':modules:ingest-geoip:internalClusterTest' --tests "org.elasticsearch.ingest.geoip.ReloadingDatabasesWhilePerformingGeoLookupsIT.test" -Dtests.seed=12714BFB74C76F6 -Dtests.security.manager=true -Dtests.locale=es-BO -Dtests.timezone=Asia/Baghdad -Druntime.java=8`

**Reproduces locally?**: Yes

**Applicable branches**: Only 7.x so far

**Failure history**: [First failure](https://build-stats.elastic.co/app/kibana#/discover?_g=(refreshInterval:(pause:!t,value:0),time:(from:now-24h,mode:quick,to:now))&_a=(columns:!(_source),index:b646ed00-7efc-11e8-bf69-63c8ef516157,interval:auto,query:(language:lucene,query:ReloadingDatabasesWhilePerformingGeoLookupsIT),sort:!(process.time-start,desc))), but this test was not merged long ago.

**Failure excerpt**:

```
2> java.lang.AssertionError: |  
-- | --
  | Expected: null |  
  | but: was <java.lang.AssertionError: |  
  | Expected: not sameInstance(<org.elasticsearch.ingest.geoip.DatabaseReaderLazyLoader@66e20dc2>) |  
  | but: was <org.elasticsearch.ingest.geoip.DatabaseReaderLazyLoader@66e20dc2>> |  
  | at __randomizedtesting.SeedInfo.seed([12714BFB74C76F6:89732B6519B01B0E]:0) |  
  | at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18) |  
  | at org.junit.Assert.assertThat(Assert.java:956) |  
  | at org.junit.Assert.assertThat(Assert.java:923) |  
  | at org.elasticsearch.ingest.geoip.ReloadingDatabasesWhilePerformingGeoLookupsIT.test(ReloadingDatabasesWhilePerformingGeoLookupsIT.java:143)
```

--- a/modules/ingest-geoip/qa/file-based-update/build.gradle
+++ b/modules/ingest-geoip/qa/file-based-update/build.gradle
@@ -11,6 +11,7 @@ apply plugin: 'elasticsearch.rest-test'
 
 testClusters.all {
   testDistribution = 'DEFAULT'
+  setting 'resource.reload.interval.high', '100ms'
 }
 
 tasks.named("integTest").configure {

--- a/modules/ingest-geoip/src/internalClusterTest/java/org/elasticsearch/ingest/geoip/ReloadingDatabasesWhilePerformingGeoLookupsIT.java
+++ b/modules/ingest-geoip/src/internalClusterTest/java/org/elasticsearch/ingest/geoip/ReloadingDatabasesWhilePerformingGeoLookupsIT.java
@@ -10,6 +10,7 @@ package org.elasticsearch.ingest.geoip;
 
 import org.elasticsearch.common.network.InetAddresses;
 import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.unit.TimeValue;
 import org.elasticsearch.common.util.concurrent.AtomicArray;
 import org.elasticsearch.index.VersionType;
 import org.elasticsearch.ingest.IngestDocument;
@@ -48,7 +49,8 @@ public class ReloadingDatabasesWhilePerformingGeoLookupsIT extends ESTestCase {
      */
     public void test() throws Exception {
         ThreadPool threadPool = new TestThreadPool("test");
-        ResourceWatcherService resourceWatcherService = new ResourceWatcherService(Settings.EMPTY, threadPool);
+        Settings settings = Settings.builder().put("resource.reload.interval.high", TimeValue.timeValueMillis(100)).build();
+        ResourceWatcherService resourceWatcherService = new ResourceWatcherService(settings, threadPool);
         try {
             final Path geoIpDir = createTempDir();
             copyDatabaseFiles(geoIpDir);

--- a/modules/ingest-geoip/src/test/java/org/elasticsearch/ingest/geoip/LocalDatabasesTests.java
+++ b/modules/ingest-geoip/src/test/java/org/elasticsearch/ingest/geoip/LocalDatabasesTests.java
@@ -11,6 +11,7 @@ package org.elasticsearch.ingest.geoip;
 import com.maxmind.geoip2.model.CityResponse;
 import org.elasticsearch.common.network.InetAddresses;
 import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.unit.TimeValue;
 import org.elasticsearch.test.ESTestCase;
 import org.elasticsearch.threadpool.TestThreadPool;
 import org.elasticsearch.threadpool.ThreadPool;
@@ -33,7 +34,8 @@ public class LocalDatabasesTests extends ESTestCase {
     @Before
     public void setup() {
         threadPool = new TestThreadPool(LocalDatabases.class.getSimpleName());
-        resourceWatcherService = new ResourceWatcherService(Settings.EMPTY, threadPool);
+        Settings settings = Settings.builder().put("resource.reload.interval.high", TimeValue.timeValueMillis(100)).build();
+        resourceWatcherService = new ResourceWatcherService(settings, threadPool);
     }
 
     @After

